### PR TITLE
Allow SummaryList rows value to be a ReactNode and include actions on…

### PR DESCRIPTION
**Pull request by LBHELewis**
… SummaryList component

### CODE REVIEW

File: src/components/SummaryList.test.tsx [hunk 1]
```diff
@@ -2,15 +2,33 @@ import React from "react";
...
+      className="class1"
+      href="/"
+      target="_blank"
+      aria-label="test link"
+      data-test="test data"
+    >
+      Link
+    </Link>
+  );
+  const links = [link];
```
Comment (erbridge) [inline src/components/SummaryList.test.tsx]: Can we put 2 links in here to show how that renders, please?

File: src/components/SummaryList.tsx [hunk 2]
```diff
@@ -51,23 +55,49 @@ export const SummaryList: React.FunctionComponent<SummaryListProps> = (
 
   const extraAttributes = Attributes.ariaAndData(props);
 
+  const anyRowHasActions = rows.some(
+    row => row.actions && row.actions.length > 0
```
Comment (erbridge) [inline src/components/SummaryList.tsx]: Tip for next time:

```suggestion
    row => row.actions?.length
```

That would do the job.

File: src/components/SummaryList.tsx [hunk 3]
```diff
@@ -19,7 +18,12 @@ export interface SummaryListRow {
   /**
    * The piece of information itself.
    */
-  value: string;
+  value: React.ReactNode;
+
+  /**
+   * A set of actions ({@link Link}) that the user can take on this information.
```
Comment (erbridge) [inline src/components/SummaryList.tsx]:
```suggestion
   * A set of actions ({@link Link|Links}) that the user can take on this information.
```

File: src/components/SummaryList.tsx [hunk 4]
```diff
@@ -51,23 +55,49 @@ export const SummaryList: React.FunctionComponent<SummaryListProps> = (
...
+                      {{ action }}
+                    </li>;
+                  })}
+                </ul>
+              )}
+            </dd>
+          ) : (
+            anyRowHasActions && (
+              <span className="govuk-summary-list__actions"></span>
+            )
```
Comment (erbridge) [inline src/components/SummaryList.tsx]: This seems a little weird. Why not use an empty `<dd>`? That would mean we could flatten some of this to be more like:

```jsx
{anyRowHasActions && <dd>{row.actions?.length && (row.actions.length === 1 ? row.actions[0] : <ul>...</ul>)}</dd>}
```

File: src/components/SummaryList.tsx [hunk 5]
```diff
@@ -51,23 +55,49 @@ export const SummaryList: React.FunctionComponent<SummaryListProps> = (
...
+              {row.actions.length == 1 ? (
+                row.actions[0]
+              ) : (
+                <ul className="govuk-summary-list__actions-list">
+                  {row.actions.map((action, index) => {
+                    <li
+                      key={index}
+                      className="govuk-summary-list__actions-list-item"
+                    >
+                      {{ action }}
```
Comment (erbridge) [inline src/components/SummaryList.tsx]:
```suggestion
                      { action }
```

Nunjucks slipping in there :).

File: CHANGELOG.md [hunk 6]
```diff
@@ -8,6 +8,12 @@ and this project adheres to
 
 ## [Unreleased]
 
+### Added
+
+- Updated SummaryList so that a row's value can be a ReactNode rather than just
+  a string
+- Updated SummaryList to allow actions to appear in summary lists
```
Comment (erbridge) [inline CHANGELOG.md]: I'm debating whether either of these belongs under a `Changed` heading... I guess they're adding features to existing features, so `Added` is probably correct.

**Comment by LBHELewis** [comment]
@erbridge All updated, thanks

**Review by erbridge — APPROVED** [review]
Nice.

You might need to kick CircleCI to get the tests to pass. Looks like something is causing memory issues.

**Pull request merged**